routes/heartbeat.go

```diff
@@ -41,13 +43,20 @@ func (h *HeartbeatHandler) ApiPost(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	rules, err := h.customRuleSrvc.GetCustomRuleForUser(user.ID)
```

Comment:
Every new heartbeat will cause a database request to fetch user rules. This seems a bit inefficient. We should use caching here. At some other part of the application, we are using `go-cache` anyway, so I'd be happy if we could utilize it here as well.

Reply:
I changed the customRuleService to use the cache before doing a database request, this way we can easily remove cached rules when they are modified.